### PR TITLE
[Netmanager] Updates polluntant concentration label on the map

### PR DESCRIPTION
**Pull request by Codebmk**
#### Summary of Changes (What does this PR do?)

- Updates the concentration label when the user clicks to change the concentration label(pm2.5/pm10).

#### Status of maturity (all need to be checked before merging):
- [x] I've tested this locally
- [x] I consider this code done
- [x] This change ready to hit production in its current state
- [x] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] I've included issue number in the "Closes #ISSUE-NUMBER" part of the "[What are the relevant tickets?](#what-are-the-relevant-tickets)" section to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [ ] I've updated corresponding documentation for the changes in this PR.
- [ ] I have written unit and/or e2e tests for my change(s).

#### How should this be manually tested?

- git pull origin staging
- git checkout ft-toggle-air-concentration
- npm run stage

#### What are the relevant tickets?

- [AN-194](https://airqoteam.atlassian.net/browse/AN-194?atlOrigin=eyJpIjoiZmU5NGIyOTkyNmU1NDFhMzgwZmRmZTgyMDcxOWE4NmEiLCJwIjoiaiJ9)

### CODE REVIEW

File: netmanager/src/views/pages/Map/OverlayMap.js [hunk 1]
```diff
@@ -127,7 +118,7 @@ const PollutantSelector = ({ className, onChange, showHeatMap }) => {
   useInitScrollTop();
   const orgData = useOrgData();
   const [open, setOpen] = useState(false);
-  const [pollutant, setPollutant] = useState("pm2_5");
+  let pollutant = localStorage.pollutant;
```
Comment (Codebmk) [inline netmanager/src/views/pages/Map/OverlayMap.js:121]: Pollutant value is stored and fetched from localstorage.

File: netmanager/src/views/pages/Map/OverlayMap.js [hunk 2]
```diff
@@ -143,47 +134,54 @@ const PollutantSelector = ({ className, onChange, showHeatMap }) => {
...
   const onHandleClick = () => {
     setOpen(!open);
   };
 
   const handleMenuItemChange = (pollutant, state) => () => {
-    setPollutant(pollutant);
+    localStorage.pollutant = pollutant;
     setOpen(!open);
     onChange(state);
+    window.location.reload();
```
Comment (Codebmk) [inline netmanager/src/views/pages/Map/OverlayMap.js:148]: Reloads the entire window once the user clicks to change pollutant value

Comment (Baalmart) [reply netmanager/src/views/pages/Map/OverlayMap.js:148]: Thanks @Codebmk for the changes. This is amazing and much welcome!

This is amazing!

I am having some reservations with the reload of the entire page whenever a user changes the concentration levels. I find it to be a somehow bad user experience.

Could we explore the possibilities that do not require the refresh of the entire page/map?

**Review by Baalmart — COMMENTED** [review]
Thanks @Codebmk , I like that we are using review comments to guide the reviewer. Much much appreciated.

I have left behind a few comments for your review.

File: netmanager/src/views/pages/Map/OverlayMap.js (hunk 1 above)
Comment (FaithDaka) [reply netmanager/src/views/pages/Map/OverlayMap.js:121]: @Codebmk When is this pollutant value removed?  Or does it stay in the client's localstorage forever?

Comment (Codebmk) [reply netmanager/src/views/pages/Map/OverlayMap.js:121]: @FaithDaka I have addressed this

File: netmanager/src/utils/setAuthToken.js [hunk 3]
```diff
@@ -1,35 +1,36 @@
...
 
       return { ...config, params: { tenant, ...params } };
     });
   } else {
     // Delete auth header
-    delete axios.defaults.headers.common["Authorization"];
+    delete axios.defaults.headers.common['Authorization'];
     localStorage.removeItem('mapStyle');
     localStorage.removeItem('mapMode');
+    localStorage.removeItem('pollutant');
```
Comment (Codebmk) [inline netmanager/src/utils/setAuthToken.js:33]: Polluntant is removed from local storage once user signs out

**Review by Baalmart — COMMENTED** [review]
Thanks @Codebmk 

I am still debating about the real value of loading the entire map just for change in a concentration value.

I will continue to appreciate and share feedback soon.

File: netmanager/src/views/pages/Map/OverlayMap.js (hunk 1 above)
Comment (Baalmart) [reply netmanager/src/views/pages/Map/OverlayMap.js:121]: A quick one @Codebmk , this deletion of items in the local storage, wasn't this functionality already present whenever a user logs out of the platform?

Thanks for the works.

Comment (Codebmk) [reply netmanager/src/views/pages/Map/OverlayMap.js:121]: Yes it is, but for every new item added, you need to explicitly define those items to remove them

**Pull request merged**